src/main/java/de/teamlapen/vampirism/entity/player/actions/ActionHandler.java

```diff
@@ -348,12 +417,17 @@ public boolean updateActions() {
             ILastingAction<T> action = (ILastingAction<T>) RegUtil.getAction(entry.getKey());
             assert action != null;
             if (newtimer == 0) {
-                action.onDeactivated(player);
-                cooldownTimers.put(entry.getKey(), action.getCooldown(player));
+                deactivateAction(action, true);
+                if(!cooldownTimers.containsKey(entry.getKey())) {
+                    cooldownTimers.put(entry.getKey(), modifiedCooldownTimers.getInt(RegUtil.id(action)));
+                }
                 it.remove();//Do not access entry after this
+
                 dirty = true;
             } else {
-                if (action.onUpdate(player)) {
+                boolean shouldDeactivate= action.onUpdate(player);
+                boolean shouldReallyDeactivate = VampirismEventFactory.fireActionUpdateEvent(player, action, newtimer, shouldDeactivate);
+                if (shouldReallyDeactivate) {
```

Comment:
I agree with that, probably the better approach. I'm not sure if there's that much of a use case when the return value of `onUpdate` needs to be known. If it returns true, it will be deactivated anyway which could be checked with the deactivation event, although I suppose it may be useful to know if the action is going to continue.

Reply:
I added a deactivate parameter to the update event